packages_rs/nextclade/src/io/gff3.rs

```diff
@@ -53,10 +54,21 @@ fn read_gff3_file_impl<P: AsRef<Path>>(filename: &P) -> Result<GeneMap, Report>
     .map(to_eyre_error)
     .collect::<Result<Vec<GffRecord>, Report>>()?;
 
-  records
+  let genemap_result = records
     .iter()
     .filter_map(convert_gff_record_to_gene_map_record)
-    .collect::<Result<GeneMap, Report>>()
+    .collect::<Result<GeneMap, Report>>();
+
+  if let Ok(genemap) = &genemap_result {
+    if genemap.len() == 0 && records.len() != 0 {
+      warn!(
+        "No valid gene entries found in genemap with {} records. No genes will be used.",
+        records.len()
+      );
+    }
+  }
+
+  genemap_result
```

Comment:
This snippet is also a bit confusing and verbose.
You typically want to get rid of `Result` and return as soon as possible, to not carry arround the checks, so something like this:

```
let genemap = (...).collect()?; // get rid of Result
if genemap.len() { ... } // no check needed
Ok(genemap)
```